plugins/woocommerce/src/Internal/Loggers/RemoteLogger.php

```diff
@@ -84,6 +276,57 @@ private function is_latest_woocommerce_version() {
 		return version_compare( WC()->version, $latest_wc_version, '>=' );
 	}
 
+	/**
+	 * Check if the error exclusively contains third-party stack frames for fatal-errors source context.
+	 *
+	 * @param string $message The error message.
+	 * @param array  $context The error context.
+	 *
+	 * @return bool
+	 */
+	protected function is_third_party_error( $message, $context ) {
+		// Only check for fatal-errors source context.
+		if ( ! isset( $context['source'] ) || 'fatal-errors' !== $context['source'] ) {
+			return false;
+		}
+
+		// If backtrace is not available, we can't determine if the error is third-party. Log it for further investigation.
+		if ( ! isset( $context['backtrace'] ) || ! is_array( $context['backtrace'] ) ) {
+			return false;
+		}
+
+		$wc_plugin_dir = 'plugins/' . dirname( WC_PLUGIN_BASENAME );
```

Comment:
- Should we trailing slash this?
- Should we normalize?
- Should we think about non-standard plugin directories?

On the first point, I'm thinking that the current logic will match paths for any plugins with a path of `plugins/woocommerce-*` (there are a huge number of these, if you scan through the slugs registered in the [plugin directory](https://plugins.svn.wordpress.org/), and there are doubtless others from marketplaces like our own and other sources).

On the second point, I just want to ensure we aren't going to have issues on Windows servers, where perhaps we fail to match because of a difference in directory separators (unless perhaps any paths inside the error message have already been normalized).

Lastly, the plugin directory is not always `wp-content/plugins` (and indeed the content directory may not be `wp-content`, though that isn't the level we're working with here).

Reply:
These are really good points. 👍  I agree that we should handle trailing slashes and normalize paths. I'll see how we can do that.

Reply:
Ok, I've updated the code to address these points in e43d48f0ed3a1a21eb4f0afe106d8c8a1d629ded. Thanks again for bringing them up! 🙏